modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/UserNotificationTypeTest.java

```diff
@@ -86,6 +91,54 @@ public void tearDown() throws PortalException {
 			user2.getUserId());
 	}
 
+	@Test
+	public void testIsDelivery() throws Exception {
+		NotificationTemplate notificationTemplate =
+			notificationTemplateLocalService.addNotificationTemplate(
+				_createNotificationContext(
+					Arrays.asList(
+						NotificationRecipientSettingUtil.
+							createNotificationRecipientSetting(
+								"userScreenName", user1.getScreenName()),
+						NotificationRecipientSettingUtil.
+							createNotificationRecipientSetting(
+								"userScreenName", user2.getScreenName())),
+					NotificationRecipientConstants.TYPE_USER));
+
+		_userNotificationDeliveryLocalService.addUserNotificationDelivery(
+			user1.getUserId(), childObjectDefinition.getPortletId(),
+			_classNameLocalService.getClassNameId(
+				childObjectDefinition.getClassName()),
+			UserNotificationDefinition.NOTIFICATION_TYPE_UPDATE_ENTRY,
+			UserNotificationDeliveryConstants.TYPE_WEBSITE, false);
+
+		_userNotificationDeliveryLocalService.addUserNotificationDelivery(
+			user2.getUserId(), childObjectDefinition.getPortletId(),
+			_classNameLocalService.getClassNameId(
+				childObjectDefinition.getClassName()),
+			UserNotificationDefinition.NOTIFICATION_TYPE_UPDATE_ENTRY,
+			UserNotificationDeliveryConstants.TYPE_WEBSITE, true);
+
+		_executeNotificationObjectAction(
+			notificationTemplate, childObjectDefinition, group.getGroupKey(),
+			user1);
+
+		_executeNotificationObjectAction(
+			notificationTemplate, childObjectDefinition, group.getGroupKey(),
+			user2);
```

Comment:
@mariuo in this case we do not need to execute the notification for each user, because the list is recipients defined at [this point](https://github.com/liferay-bpm/liferay-portal/pull/4547/files#diff-40384576a458f8a699d6aa18667a2772d50acc545463c64a3572b4ee793a81e0R99-R105) already contains the users to be notified (user1 and user2).